src/components/list/list.mdx

```diff
@@ -8,23 +8,31 @@ menu: Components
 
 <limel-props name="limel-list" />
 
+## Custom styles
+
+| Class name | Description                                                                                     |
+| -------- | -------------------------------------------------------------------- |
+| `has-striped-rows` | Adds distinct styling and more interactivity to list items |
```

Comment:
Do you think we need  `has` in the name? :thinking: 
Maybe `striped-rows` is enough?

Reply:
Yes I think having such words makes the classes more "descriptive" and easier to understand, specially for people who don't know our intentions of having such a class. Personally I'd recommend having them. But open for discussion.

some examples for this naming convention:

`dark` vs. `has-dark-background`
`disabled` vs. `is-disabled`
`dimmed` vs. `gets-dimmed-while-loading`
`hover-zoom` vs. `gets-zoomed-on-hover`

Reply:
Being short, abstract and concise is sometimes not that good. In these cases I think descriptive class names makes the code more understandable.

Reply:
For me it does not have any meaning in this case and is just clutter :smile: Since you have added the class you would expect it to be applied. We could flip the table and look at it another way, having a class named `has-not-striped-rows` does not make much sense, but is almost the same thing
We are all different I suppose but this is my view on it, you can keep it or change it if you want :slightly_smiling_face:

Reply:
`dark` vs. `has-dark-background`

I think `dark-background` is more informative than `dark`, but I don't think `has-dark-background` is more informative than `dark-background`. The "has" is conveyed by the fact that the class is applied.

`disabled` vs. `is-disabled`

Same thing here. If it's not disabled, then the class just isn't there.

The last two examples have more merit, in my opinion, but I would skip the `gets-` for the same reason as above.

`dimmed-while-loading` and `zoomed-on-hover` are expressive, but not unnecessarily verbose, in my opinion 🙂

Reply:
I understand what you mean, and I'm not trying to defend the class name. Mainly want to pick your brain on the subject. 

Now my examples weren't so well-thought 😛. It takes some time to find good examples. But imagine you see this imaginary code:

`<button class="lime-button is-in-top-header has-accented-background-color has-icon-left gets-fancy-effect-on-hover" ····`
versus
`<button class="lime-button top-header accented-background icon-left fancy-effect" ····`

For me personally, it feels I'm reading a sentence that describes how the button would look and behave like. I can more quickly build an assumption around the styles and expect things. Also expect for example that the creators have a solid idea regarding buttons which are located in the top header -they must be behaving or looking differently compared to similar buttons-; or there is a visual effect that only gets applied on hover not in other states - so if I want to inspect it I must hover. And so forth... In the minimal example, you almost have to find the classes and read their respective styles to build this understanding.

I don't know...., I kindda like this type of naming convention more, as a UX designer. But I totally also understand your points, and know that abbreviations and short-codes and so forth are always more desired in coding mindset. And of course I'm aware that too long words make reading the sentence also too hard eventually.

I also think the same way for class-name conventions that describe the structure. For example seeing this class `container` will immediately 😵💀me. I would wonder container of what you ☀️of a 🏖?! 😄 Why don't you specify more? ’button-group-container`? `top-menu-container`? can't you put something that is easy to track/understand/target? 😜

Anyhow, I have no good bullet-proof argument or naming convention to recommend. Open for suggestions, change, new conventions, anything you guys say.

Reply:
`has-dark-background`
`gets-dark-background-when-focused`
`applies-dark-background-to-something`
`buys-dark-background-for-dinner-😁`

Reply:
Yeah, good points I think.

> abbreviations and short-codes and so forth are always more desired in coding mindset

I'm actually mostly against using abbreviations. They should only be used when it helps the *reader*, not the *writer*.

I'm fine with either `has-striped-rows` or `striped-rows`. I didn't want to but in on the decision, just add my thoughts to the conversation 👍🙂

Reply:
Totally right. Abbreviations should be forbidden, unless everybody knows what you are talking about, e.g. SMS or Covid-19. Nobody knows what it stands for, but everyone know what it is. 

As you beautifully said, it helps the reader!

A bit more thoughts on the topic...
Since a class name such as `striped-rows` can be also perceived as "being" something (being striped rows [maybe striped rows is a component in itself, for instance `is-striped-rows`]), it's not favorable. 

But `has-striped-rows` automatically includes and expresses a few things about the component: 
1: whatever this thing is, it has rows
2: and the rows are striped
3-ish: it probably can have rows which aren't striped too

So, I get a feeling that we're almost on the same side. I suggest sticking to this convention :)